test/dutchExchange-claimBuyerFunds.js

```diff
@@ -0,0 +1,347 @@
+      0,
+      2,
+      1,
+      { from: seller1 },
+    )
+
+    eventWatcher(dx, 'Log', {})
+  })
+
+  after(eventWatcher.stopWatching)
+
+  it('1. check for a throw, if auctionIndex is bigger than the latest auctionIndex', async () => {
+    const auctionIndex = await getAuctionIndex()
+    await setAndCheckAuctionStarted(eth, gno)
+    await waitUntilPriceIsXPercentOfPreviousPrice(eth, gno, 1.5)
+    await assertRejects(dx.claimBuyerFunds(eth.address, gno.address, buyer1, auctionIndex + 1))
+  })
+
+  it(' 2. checks that the return value == 0, if price.num ==0 ', async () => {
+    // prepare test by starting and clearning new auction
+    let auctionIndex = await getAuctionIndex()
+    await postSellOrder(gno, eth, 0, totalSellAmount2ndAuction, seller2)
+    await waitUntilPriceIsXPercentOfPreviousPrice(eth, gno, 1)
+    await postBuyOrder(eth, gno, auctionIndex, 2 * 10e18, buyer1)
+    auctionIndex = await getAuctionIndex()
+    await setAndCheckAuctionStarted(eth, gno)
+    assert.equal(2, auctionIndex)
+
+    // now claiming should not be possible and return == 0
+    await setAndCheckAuctionStarted(eth, gno)
+    await waitUntilPriceIsXPercentOfPreviousPrice(eth, gno, 1.5)
+    const [closingPriceNum] = (await dx.closingPrices.call(gno.address, eth.address, auctionIndex - 1)).map(i => i.toNumber())
+    assert.equal(closingPriceNum, 0)
+    const [claimedAmount] = (await dx.claimBuyerFunds.call(gno.address, eth.address, buyer1, auctionIndex)).map(i => i.toNumber())
+    assert.equal(claimedAmount, 0) 
+  })
+  it(' 3. checks that a non-buyer can not claim any returns', async () => {
+    const auctionIndex = await getAuctionIndex()
+    const [claimedAmount] = (await dx.claimBuyerFunds.call(eth.address, gno.address, buyer1, auctionIndex)).map(i => i.toNumber())
+    assert.equal(claimedAmount, 0) 
+  })
+
+  it('4. check right amount of coins is returned by claimBuyerFunds if auction is not closed', async () => {
+    const auctionIndex = await getAuctionIndex()
+
+    // prepare test by starting and closing theoretical auction
+    await waitUntilPriceIsXPercentOfPreviousPrice(eth, gno, 1)
+
+    await postBuyOrder(gno, eth, auctionIndex, totalSellAmount2ndAuction / 4, buyer2)
+
+    // checking that closingPriceToken.num == 0
+    const [closingPriceNumToken] = (await dx.closingPrices.call(eth.address, gno.address, auctionIndex)).map(i => i.toNumber())
+    assert.equal(closingPriceNumToken, 0)
+    
+    // actual testing at time with previous price
+    const [claimedAmount] = (await dx.claimBuyerFunds.call(gno.address, eth.address, buyer2, auctionIndex)).map(i => i.toNumber())
+    const [num, den] = (await dx.getPriceForJS.call(gno.address, eth.address, auctionIndex)).map(i => i.toNumber())
+    let sellVolume = (await dx.sellVolumesCurrent.call(gno.address, eth.address))
+    let buyVolume = await dx.buyVolumes.call(gno.address, eth.address)
+    let oustandingVolume = (sellVolume.mul(bn(num)).div(den)).sub(buyVolume)
```

Comment:
You don't need `bn(num)` here, what you pas to `BN.method()` is automatically wrapped in `bn`